addons/account/models/account_move.py

```diff
@@ -1295,8 +1295,11 @@ def _check_duplicate_supplier_reference(self):
                 AND move2.id != move.id
             WHERE move.id IN %s
         ''', [tuple(moves.ids)])
-        if self._cr.fetchone():
-            raise ValidationError(_('Duplicated vendor reference detected. You probably encoded twice the same vendor bill/credit note.'))
+        duplicated_moves = self.browse([r[0] for r in self._cr.fetchall()])
+        if duplicated_moves:
+            raise ValidationError(_('Duplicated vendor reference detected. You probably encoded twice the same vendor bill/credit note:\n%s') % "\n".join(
+                duplicated_moves.mapped(lambda m: "%(partner)s - %(ref)s - %(date)s" % {'ref': m.ref, 'partner': m.partner_id.display_name, 'date': format_date(self.env, m.date)})
```

Comment:
You will show to the user the same values he encoded in the form view... I don't think it's relevant to add such values in the error message..

Reply:
Not if you upload multiple bills with the upload button